### ISSUE TITLE
复现问题

### ISSUE THREAD

**Issue by zhu2014yi** [zhanght021/RPT#7]
Hi，我按照您的论文中的描述复现了一下您的论文，但是我发现刚开始一阶段中是没有正样本的，导致第二阶段没了正样本来回归和分类，二阶段回归loss一直都是0，请问您是如何解决的。

**Comment by zhanght021** [comment]
可能论文中描述有问题，如果我没记错的话应该不管第一阶段的结果有没有iou＞0.6的样本，第二阶段都会选择iou最大的那个作为正样本。你可以去看看reppoints检测算法的代码，整个训练代码几乎和那个是一致的，它就是这么实现的

**Comment by zhu2014yi** [comment]
> 可能论文中描述有问题，如果我没记错的话应该不管第一阶段的结果有没有iou＞0.6的样本，第二阶段都会选择iou最大的那个作为正样本。你可以去看看reppoints检测算法的代码，整个训练代码几乎和那个是一致的，它就是这么实现的@zhu2014yi 本来训练代码也会立马来源。因为我们做了一点后续的工作，所以准备后续再来源，，可以参考reppoints的实现

感谢

**Comment by zorrocai** [comment]
Hi @zhu2014yi , 我也在尝试复现这篇文章。但是有个疑问，不知你们是怎么处理的。二阶段分类分支输出的logit维度为 [batch, score_size*score_size,1], 构建的标签维度为 [batch, score_size*score_size,1]， 但其自带的sigmoid_focal_loss中似乎要求logit维度为[batch, C, ...]， 第二维为类别个数。

**Comment by zhanght021** [comment]
具体你可以看一下FCOS怎么实现的https://github.com/tianzhi0549/FCOS

pred_label = cls_scores.permute(0, 2, 3, 1).view(-1, 1)
truth_label = label.view(-1).int()
cls_loss = SigmoidFocalLoss(2, 0.25)(pred_label, truth_label).sum() / label.sum()
@zorrocai

**Comment by zorrocai** [comment]
感谢
